libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -474,6 +480,13 @@ method subscribe*(g: GossipSub,
                   topic: string,
                   handler: TopicHandler) {.async.} =
   await procCall PubSub(g).subscribe(topic, handler)
+
+  # add our current peers to availability table
+  for peerid, _ in g.peers:
+    if topic notin g.gossipsub:
+      g.gossipsub[topic] = initHashSet[string]()
+    g.gossipsub[topic].incl(peerid)
```

Comment:
This isn't the right place to do it, the `gossipsub` table is the table that holds a map of topics to peers, with the peers being those that previously advertised that they subscribe to the topic, what this would do is wrongly map peers to topics that they aren't subscribed to.